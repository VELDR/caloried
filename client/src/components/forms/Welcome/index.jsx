import { useDispatch } from 'react-redux';
import { FormattedMessage } from 'react-intl';

import { nextStep } from '@pages/SignUp/actions';
import FormFooter from '@components/forms/FormFooter';

import classes from './style.module.scss';

const Welcome = () => {
  const dispatch = useDispatch();
  return (
    <div className={classes.container} data-testid="welcome">
      <div className={classes.body}>
        <div className={classes.title}>
          <FormattedMessage id="app_welcome_title" />
          ðŸŒ±
        </div>
        <div className={classes.subtitle}>
          <FormattedMessage id="app_welcome_subtitle" />
          ðŸŽ¯
        </div>
      </div>
      <div className={classes.footer}>
        <FormFooter onContinue={() => dispatch(nextStep())} />
      </div>
    </div>
  );
};

export default Welcome;
