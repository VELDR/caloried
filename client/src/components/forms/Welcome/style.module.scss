@import '@styles/base';

.container {
  @include flex-center-column(3rem);
  height: 100%;

  .body {
    @include flex-center-column(1rem);
    text-align: center;
    .title {
      font-size: $font-size-larger;
      font-weight: 700;
    }
    img {
      @include box(12rem);
    }
    .subtitle {
      @include apply-at-md-size {
        width: 75%;
      }
    }
  }
}
