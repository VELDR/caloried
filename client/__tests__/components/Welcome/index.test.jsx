import Welcome from '@components/forms/Welcome';
import { render } from '@utils/testHelper';

let wrapper;

beforeEach(() => {
  wrapper = render(<Welcome />);
});

describe('Welcome Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('welcome')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
