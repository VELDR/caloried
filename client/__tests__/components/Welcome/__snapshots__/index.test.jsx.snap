// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="welcome"
      >
        <div
          class="body"
        >
          <div
            class="title"
          >
            Langkah Menuju Hidup Sehat
            ðŸŒ±
          </div>
          <div
            class="subtitle"
          >
            Sebelum memulai, kami ingin mengetahui lebih tentang Anda untuk membuat pengalaman yang disesuaikan dengan tujuan Anda.
            ðŸŽ¯
          </div>
        </div>
        <div
          class="footer"
        >
          <div
            class="footer"
            data-testid="form-footer"
          >
            <button
              class="button footer__right footer__button"
              data-testid="primary-button"
              type="submit"
            >
              Mulai
               
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EastIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="welcome"
    >
      <div
        class="body"
      >
        <div
          class="title"
        >
          Langkah Menuju Hidup Sehat
          ðŸŒ±
        </div>
        <div
          class="subtitle"
        >
          Sebelum memulai, kami ingin mengetahui lebih tentang Anda untuk membuat pengalaman yang disesuaikan dengan tujuan Anda.
          ðŸŽ¯
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="footer"
          data-testid="form-footer"
        >
          <button
            class="button footer__right footer__button"
            data-testid="primary-button"
            type="submit"
          >
            Mulai
             
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EastIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
